### PR TITLE
Bugfix for string replace.

### PR DESCRIPTION
**Pull request by robertsipka**
Some test case were failed with enabled show-opcodes build option.
For example:
"98765".replace(76);
print ("\ud801\udc00".replace("\ud801", "#") === "#\udc00");

JerryScript-DCO-1.0-Signed-off-by: Robert Sipka rsipka.uszeged@partner.samsung.com

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.c [hunk 1]
```diff
@@ -875,55 +875,65 @@ ecma_builtin_string_prototype_object_replace_get_string (ecma_builtin_replace_se
     {
       ecma_char_t action = LIT_CHAR_NULL;
 
-      if (*replace_str_curr_p++ == LIT_CHAR_DOLLAR_SIGN)
+      if (*replace_str_curr_p != LIT_CHAR_DOLLAR_SIGN)
       {
-        if (replace_str_curr_p < replace_str_end_p)
+      /* if not a continuation byte */
+        if ((*replace_str_curr_p & LIT_UTF8_EXTRA_BYTE_MASK) != LIT_UTF8_EXTRA_BYTE_MARKER)
```
Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtin-string-prototype.c]: Is this comment properly aligned?

**Comment by zherczeg** [comment]
I think we should mention the failing test in the commit message

**Comment by robertsipka** [comment]
Thanks for the review, I've updated the PR.

**Comment by zherczeg** [comment]
LGTM

**Comment by robertsipka** [comment]
Thanks, I've rebased this PR.

**Pull request merged**